wikitexthtml/render/preprocess.py

```diff
@@ -18,7 +18,7 @@
 ONLYINCLUDE_PATTERN = regex.compile(r"<onlyinclude>([\s\S]*?)(?></onlyinclude>|\Z)")
 PRE_PATTERN = regex.compile(r"<pre>([\s\S]*?)(?></pre>|\Z)")
 SYNTAXHIGHLIGHT_PATTERN = regex.compile(
-    r"<syntaxhighlight( lang=\")?(?P<lang>[a-zA-z]+)?\"?"
+    r"<syntaxhighlight( lang=\")?(?P<lang>[^\"]+)?\"?"
```

Comment:
While at it, there is something else strange about this regex .. take the following example:

`<syntaxhighlight c++ line 12>`

This is selecting the language `c++ line 12`. On the other hand:

`<syntaxhighlight c++" line 12>`

Does the same as:

`<syntaxhighlight lang="c++" line 12>`

A possible solution is to drop `lang=` completely, or otherwise make it an option: either `lang="c++"` or `c++`. Or don't support `c++` (without `lang`) syntax at all.

Not sure if you picked an existing syntax for enabling the highlighter or that you made this syntax up, but looking at how most (not all) of mediawiki works, one could also opt for having `lang="c++" line=12` as only valid syntax.

What do you think?

Reply:
Ah, so it is existing! And `start=` got left out ;) Let me see if I can help a bit with this regex ... I hate regex :P

Reply:
(only difference is that I made `line=""` part of a group, so you cannot have `lang` captured without having `lang=""` there. And use `start=` instead of just a number ;))

Reply:
I suggest to take a look at my regex above ;) Yours still allows for `<syntaxhighlight java" line start=12>blashf</syntaxhighlight>` as input, which I would consider buggy ;)

Or maybe put differently: why didn't you capture `lang` together with the `lang=""` part, but why use a 3-part which all 3 parts optional?

Reply:
In a nice linky: https://regex101.com/r/UPfWCA/1

Reply:
> why didn't you capture lang together with the lang="" part, but why use a 3-part which all 3 parts optional?

Because I don't do regex and built this linearly over the course of months, that's why 🙃

Reply:
Always nice to know I am not alone :P Regex are the worst .. reason I am so happy with https://github.com/5j9/wikitextparser .. he did all the complicated regex for us :D

If you like what you see in my link, I suggest to just use my regex. But feel free to fiddle with it yourself .. it is an endless battle :P

Reply:
Yeah works for me, I pushed that along with another tweak to the render() method.